component/src/main/ballerina/ballerina/net/reliable/processor/processor.bal

```diff
@@ -0,0 +1,100 @@
+    if (guaranteedProcessor != null) {
+        log:printWarn("processor is already registered and running");
+        return false;
+    }
+    guaranteedProcessor = processor;
+
+    //start the task
+    function () returns (error) onTriggerFunction = handleMessage;
+
+    function (error e) onErrorFunction = cleanupError;
+
+    var taskId, schedulerError = task:scheduleTimer(onTriggerFunction,
+                                                    onErrorFunction, {delay:1000, interval:guaranteedProcessor.interval});
+    guaranteedProcessor.taskId = taskId;
+    retryCounterMap[taskId] = guaranteedProcessor.retryCount;
+    if (schedulerError != null) {
+        log:printError("task schduling failed: " + schedulerError.msg) ;
+    } else {
+        log:printDebug("task schduled with ID: " + taskId);
+    }
+    return true;
+}
+
+function handleMessage() returns (error) {
+    int retryIteration;
+    retryIteration,_ = (int) retryCounterMap[guaranteedProcessor.taskId];
+
+    string destinationName = guaranteedProcessor.destinationName;
+
+    blob retrievedMessageStream;
+    try {
+        transaction with retries(0) {
+            retrievedMessageStream = guaranteedProcessor.retrieve(guaranteedProcessor.config, destinationName);
+            string empty = "";
+            if (retrievedMessageStream.toString("UTF-8") != empty) {
+                log:printDebug("message received at the processor");
+                //retrying is over
+                if (retryIteration == 0) {
+                    retryCounterMap[guaranteedProcessor.taskId] = guaranteedProcessor.retryCount;
+                    log:printWarn("moving message to dlc due to max retry exceeded");
+                    guaranteedProcessor.store(guaranteedProcessor.config, destinationName + "_dlc", retrievedMessageStream);
+                } else {
+                    error e = guaranteedProcessor.handler(retrievedMessageStream);
+                    if (e != null) {
+                        log:printError("endpoint invocation failed for " + (guaranteedProcessor.retryCount - retryIteration + 1) + " iteration");
+                        retryCounterMap[guaranteedProcessor.taskId] = retryIteration - 1;
+                        // catching error here and handling is not possible at the moment, due to an issue in
+                        // ballerina transactions. When its fixed try-catch, throwing the error
+                        // can be removed and we can do a transaction 'abort' after catching the error
+                        throw e;
+                    }
+                }
+            }
+            retryCounterMap[guaranteedProcessor.taskId] = guaranteedProcessor.retryCount;
+        }
+    } catch (error e) {
+        // The log and try-catch block can be removed after fixing the the issue
+        // https://github.com/ballerinalang/ballerina/issues/4322
+        log:printDebug("error while transaction is caught. Rollback the transaction.");
+    }
+    return null;
```

Comment:
Should the error be always null?

Reply:
Handled the errors inside the method and returned null, as the task should not break upon an error.

Reply:
Ack